### ISSUE TITLE
Settings can not be accessed

### ISSUE THREAD

**Issue by pfirsich** [footballhead/love-launcher#3]
I'm using Atom 0.182.0 on Windows and it seems like that there is no way to graphically access the settings of love-launcher. Clicking on the "settings" button in the "Packages" overview only yields a summary of the shortcuts (only being the launch shortcut).

**Comment by footballhead** [comment]
Hi there, thanks for the report!

This is an issue and it should affect all platforms: Atom has changed the way that configuration is handled and I'm still using the old way. [Here's the relevant blog post.](http://blog.atom.io/2014/10/02/config-api-has-schema.html) I can image that they're slowly fazing out the deprecated stuff and this is the version that broke the camels back.

I've tried modifying the code to use the new API but I'm a few hours in with no results. But I am looking into this issue and aim to have a fix within the next 24 hours.

**Comment by pfirsich** [comment]
It doesn't seem like you changed anything, but it works now!

**Comment by footballhead** [comment]
That's... incredibly weird. Yeah I haven't changed anything yet. And I still can't get it to work on my end... I may need to take it to the Atom forums because this is driving me insane.

**EDIT:** Going to keep track of a list of things I've tried:
- Rebooting
- Restarting Atom
- Uninstalling and installing the package

**Comment by footballhead** [comment]
FOUND IT!

The settings only appear _after_ running **Love Launcher: Launch** at least once.

Repro:
0. Close atom if already open
1. Open atom
2. Check love launcher settings; there should be nothing there
3. Close the settings tab
4. Open a file; it can be new, already existing, whatever
5. Run **Love Launcher: Launch** (ALT-L) [edit: put in the _right_ keymap :P]
6. Check love launcher settings; the settings should now show up.

This probably means that the script isn't being properly run on boot so the configuration isn't being set. That could be a misconfiguration on my part but I'm not sure.

But this is a start.

**Comment by footballhead** [comment]
The problem is several fold as I understand it:
1. The old API used the `activationEvents` property in `package.json` to load necessary settings. [This is the document that I was most likely reading as I made this package](https://atom.io/docs/v0.64.0/your-first-package); note the use of `activationEvents` under _Trigger the Command_. **The new API uses `activationCommands` instead.** [See the most recent version of the document.](https://atom.io/docs/v0.184.0/your-first-package)
2. Specifying `activationCommands` delays loading until the command is needed. Which I think means my file isn't loaded until the very last minute, which means my settings aren't recognized until the command is run ([see this discussion post](https://discuss.atom.io/t/having-activationevents-is-confusing/14440/3)). However, the "not loading the settings" thing seems to be fairly new behaviour since it worked fine before.
3. I haven't invested enough time into learning coffeescript so my package still has a few bugs in it (some I just found by comparing my code to the coffeescript specs).

What's the easy fix? Just remove `activationCommands`, which will cause the file to load on startup and recognize the settings. However, this was deemed as not being a "good citizen" in that discussion post. _However_ my script on this paltry Intel Core 2 Duo with 2GB RAM takes < 30 ms to load, which isn't very much. For the short term this is probably what I'll do.
